client/components/close-on-escape/index.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { filter, includes, isEmpty, last, noop } from 'lodash';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+
+const ESC_KEY_CODE = 27;
+let components = [];
+
+function onKeydown( event ) {
+	if (
+		! isEmpty( components ) &&
+		event.keyCode === ESC_KEY_CODE &&
+		! isInput( event.target )
+	) {
+		const component = last( components );
+
+		component.onEscape();
+	}
+}
+
+function isInput( element ) {
+	return includes( [ 'INPUT', 'TEXTAREA' ], element.nodeName );
```

Comment:
When testing this change (over the original `-1 !== [ 'INPUT', 'TEXTAREA' ].indexOf( element.nodeName );`), that the modal still closes when using one of the inputs in the `media-modal`s detail view.

I commented out the call to `.onEscape` in `onKeyDown` like so, yet still saw the modal close:

```js
function onKeydown( event ) {
	if (
		! isEmpty( components ) &&
		event.keyCode === ESC_KEY_CODE &&
		! isInput( event.target )
	) {
		const component = last( components );

		component.onEscape();
	}
}
```

I'll dig in to this a little more before merging this, testing the original mixin and looking for other listeners on the <kbd>esc</kbd> key.